frontend/src/components/Charts/DepthChart/index.tsx

```diff
@@ -0,0 +1,320 @@
+      return (Number(datum.x) > center - xRange) && 
+             (Number(datum.x) < center + xRange)
+    })
+
+    return inlineSerie
+  }
+
+  const closeSerie = (serie: Datum[], limitBottom: number, limitTop: number): Datum[] =>{
+    if (serie.length == 0) { return [] }
+
+    // If the bottom is not 0, exdens the horizontal bottom line
+    if (serie[0].y !== 0) {
+      const startingPoint: Datum = {
+        x: limitBottom,
+        y: serie[0].y
+      }
+      serie.unshift(startingPoint)
+    }
+
+    // exdens the horizontal top line
+    const endingPoint: Datum = {
+      x: limitTop,
+      y: serie[serie.length - 1].y
+    }
+
+    return [...serie, endingPoint]
+  }
+
+  const formatAxisX = (value: number): string => {
+    if (xType === 'base_amount') {
+      return value.toString()
+    }
+    return `${value}%`
+  }
+  const formatAxisY = (value: number): string => `${value}BTC`
+
+  const rangeSteps = xType === 'base_amount' ? 200 : 0.5
+  
+  const generateTooltip: React.FunctionComponent<PointTooltipProps> = (pointTooltip: PointTooltipProps) => {
+    const order: Order = pointTooltip.point.data.order
+    return order ? (
+      <Paper elevation={12} style={{ padding: 10, width: 250 }}>
+        <Grid container justifyContent="space-between" xs={12}>
+        <Grid
+          xs={3}
+          container
+          direction="column"
+          justifyContent="center"
+          alignItems="flex-start"
+        >
+            <Box>
+              <RobotAvatar order={order} />
+            </Box>
+          </Grid>
+          <Grid 
+            container 
+            xs={8}
+            direction="column"
+            justifyContent="center"
+            alignItems="flex-start"
+          >
+            <Box>
+              {order.maker_nick}
+            </Box>
+            <Box>
+              <Grid
+                container
+                direction="column"
+                justifyContent="flex-start"
+                alignItems="flex-start"
+              >
+                <Grid item xs={12}>
+                  {amountToString(order.amount, order.has_range, order.min_amount, order.max_amount)}
+                  {' '}
+                  {currencyDict[order.currency]}
+                </Grid>
+                <Grid item xs={12}>
+                  <PaymentText 
+                    othersText={t("Others")} 
+                    verbose={true} 
+                    size={12} 
+                    text={order.payment_method}
+                  />
+                </Grid>
+              </Grid>
+            </Box>
+          </Grid>
+        </Grid>
+      </Paper>
+    ) : <></>
+  }
+
+  const handleOnClick: PointMouseHandler = (point: Point) => {
+    history.push('/order/' + point.data?.order?.id);
+  }
+  
+  return bookLoading || !center || enrichedOrders.length < 1 ? (
+    <div style={{display: "flex", justifyContent: "center", paddingTop: 200, height: 420 }}>
+      <CircularProgress />
```

Comment:
The chart feels sluggish to load if you switch a few times between List/Chart repeatedly. I believe this is because every single time the chart is loaded, there is a call to "/limits/" that has to be waited.

Do you think we could save `/limits/` into the app state and from that point on only update that variable? (same as we do with the book orders)

Reply:
Good catch! I change it and now it only loads once :)